src/main/services/additional-content/maps/local-maps-manager.service.ts

```diff
@@ -306,6 +309,85 @@ export class LocalMapsManagerService {
         return null;
     }
 
+    public importMaps(zipPaths: string[], version?: BSVersion): Observable<Progression> {
+        const progress: Progression<string> = {
+            total: zipPaths.length,
+            current: 0,
+            data: "",
+        };
+
+        return new Observable<Progression<string>>(observer => {
+            (async () => {
+                for (const zipPath of zipPaths) {
+                    ++progress.current;
+                    progress.data = zipPath;
+                    observer.next(progress);
+
+                    try {
+                        await this.importMap(zipPath, version);
+                    } catch (error: any) {
+                        log.error(`Could not import "${zipPath}"`, error);
+                        this.ipc.send<{map: string, error: string}>(
+                            "map-not-imported",
+                            this.windows.getWindows("index.html").at(0),
+                            { map: zipPath, error: error.code }
+                        );
+                    }
+                }
+                observer.complete();
```

Comment:
Prefer placing the .complete() in the finally block and add an .error() in the catch of the promise.\
This way, it will handle all potential errors that are not caught by your try/catch block.\
(Even though, I agree, in the current state, unhandled errors are shouldn't possible)